parsexml.py

```diff
@@ -9,39 +9,82 @@
+    dateiso = date.isoformat()+'Z'
+    return dateiso
+
 #this function contains the code for parsing the xml file
 #and writing the results out to a json file
 def processFile(fname):
+    type = os.path.splitext(fname)[1]
+    doc = {}
     try:
         with open(os.path.abspath(fname)) as fd:
-            fn = change_ext(fname,'json')
-            doc = xmltodict.parse(fd.read())
-
+            if type == ".xml":
+                fn = change_ext(fname,'json')
+                doc = xmltodict.parse(fd.read())
+            elif type == ".json":
+                doc = json.loads(fd.read())
             #get document id from metadata xml file
-            for x in doc['main']['DATA_RECORD']:
-                x['appid'] = x.pop('BD_PATENT_APPLICATION_NO')
-                x['doc_date'] = x.pop('DOCUMENT_CREATE_DT')
-                value = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
-                txtfn = os.path.join(os.path.dirname(fn),'PDF_image',value+'.txt')
-                if os.path.isfile(txtfn):
-                    with open(txtfn) as dr:
-                        text = dr.read()
-                        x['textdata'] = text
-                else:
-                    logging.error("PDF text file does not exist. JSON file creation skipped.")
-                    return
+            records = doc['main']['DATA_RECORD']
+            for x in records:
+                docid = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
+                if type == ".xml":
+                    txtfn = os.path.join(os.path.dirname(fn),'PDF_image',docid+'.txt')
+                    if os.path.isfile(txtfn):
+                        x['LAST_MODIFIED_TS'] = format_date('LAST_MODIFIED_TS',x)
+                        x['PATENT_ISSUE_DT'] = format_date('PATENT_ISSUE_DT',x)
+                        x['DECISION_MAILED_DT'] = format_date('DECISION_MAILED_DT',x)
+                        x['PRE_GRANT_PUBLICATION_DT'] = format_date('PRE_GRANT_PUBLICATION_DT',x)
+                        x['APPLICANT_PUB_AUTHORIZATION_DT'] = format_date('APPLICANT_PUB_AUTHORIZATION_DT',x)
+                        x['appid'] = x.pop('BD_PATENT_APPLICATION_NO')
+                        x['doc_date'] = x.pop('DOCUMENT_CREATE_DT')
+                        with open(txtfn) as dr:
+                            text = dr.read()
+                            x['textdata'] = text
+                    else:
+                        logging.info("TXT file: "+docid+".txt  does not exist. JSON file creation skipped.")
+                        return
+                if (args.solr):
+                    jsontext = json.dumps(x)
+                    #Send content to Solr to update index
+                    jsontext = '{"add":{ "doc":'+jsontext+',"boost":1.0,"overwrite":true, "commitWithin": 1000}}'
+                    with open(os.path.join(os.path.dirname(fname),'jsoncomplete.txt'),'a+') as logfile:
+                        logfile.seek(0)
+                        if docid+"\n" in logfile:
```

Comment:
The reason I needed to add the seek(0) is because append mode `a+` behaves differently on Linux and OS X.
http://bugs.python.org/issue22651

From Linux:
```
a+ - Open for reading and appending (writing at end of file). The file is created if it does not exist. The initial file position for reading is at the beginning of the file, but output is always appended to the end of the file.
```

From OS X:
```
"a+'' - Open for reading and writing. The file is created if it does not exist. The stream is positioned at the end of the file. Subsequent writes to the file will always end up at the then current end of file, irrespective of any intervening fseek(3) or similar.
```

I tested this in the python console too.  Even though `seek(0)` goes to the beginning of the file for reading, `f.write()` still defaults to the last line of the file.